src/puppetlabs/jdbc_util/core.clj

```diff
@@ -4,13 +4,20 @@
             [clojure.string :as str]
             [puppetlabs.kitchensink.core :as ks]))
 
+(defmacro with-timeout [timeout-s default & body]
+  `(let [f# (future (do ~@body))
+         result# (deref f# (* 1000 ~timeout-s) ~default)]
+     (future-cancel f#)
+     result#))
```

Comment:
We should find a good place to put this given that it's defined in two of our projects now. Seems like a good candidate for kitchen sink.